backend/inspirehep/migrator/tasks.py

```diff
@@ -444,15 +444,21 @@ def migrate_record_from_mirror(
         )
         prod_record.error = exc
         db.session.merge(prod_record)
-    except DownloadFileError:
-        removed_cached_files = remove_cached_afs_file_locations(original_urls)
-        if not removed_cached_files:
-            raise
-        return migrate_record_from_mirror(
-            prod_record=prod_record,
-            disable_orcid_push=disable_orcid_push,
-            disable_relations_update=disable_relations_update,
-        )
+    except DownloadFileError as exc:
+        # The DownloadFileError was not caused by an invalid cache
+        if not original_urls:
```

Comment:
`is_error_caused_by_invalid_cache = original_urls is not None` could be better just to avoid the comment.